### PR TITLE
Branding Logo UI

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
Refactoring of Image Component Settings to be reused for Branding Logo.
Initial Branding Logo implementation, not yet visible to users as no branded Templates have been released

## Motivation and Context
Branding Epic JTBDs

## How Has This Been Tested?
Manually tested on https://apps-stage-1.risevision.com/

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Comment by ezequielc** [comment]
@alex-deaconu There are some areas for improvement, like loading metadata for logo and handling branding update failures, but we can add them on top of what I have here.
Please review. Thanks!

File: web/scripts/template-editor/components/services/svc-regular-image-factory.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+angular.module('risevision.template-editor.services')
+  .factory('regularImageFactory', ['fileMetadataUtilsService', 'blueprintFactory', 'templateEditorFactory',
```
Comment (alex-deaconu) [inline web/scripts/template-editor/components/services/svc-regular-image-factory.js]: @ezequielc I think I would name this `baseImageFactory`. Thanks.

File: web/scripts/template-editor/services/svc-blueprint-factory.js [hunk 2]
```diff
@@ -24,6 +24,36 @@ angular.module('risevision.template-editor.services')
         return (!!factory.blueprintData && factory.blueprintData.branding === true);
       };
 
+      factory.getBlueprintData = function (componentId, attributeKey) {
```
Comment (alex-deaconu) [inline web/scripts/template-editor/services/svc-blueprint-factory.js]: @ezequielc I would rename this function. I realize it's inheritance from the `ctr-template-editor`, but maybe the factory function should be renamed more appropriately. Also, maybe we can separate into something like `getComponent()` and `getComponentAttribute()`, having a function with 2 return structures is not ideal. Thanks.

File: web/scripts/template-editor/components/services/svc-branding-factory.js [hunk 3]
```diff
@@ -28,6 +30,18 @@ angular.module('risevision.template-editor.services')
               accentColor: settings.brandingAccentColor
             };
           }
+
+          if (factory.brandingSettings.logoFile) {
```
Comment (alex-deaconu) [inline web/scripts/template-editor/components/services/svc-branding-factory.js]: @ezequielc can we split this off into its own function called from here? Thanks.

File: web/scripts/template-editor/components/services/svc-branding-factory.js [hunk 4]
```diff
@@ -28,6 +30,18 @@ angular.module('risevision.template-editor.services')
...
+            fileExistenceCheckService.requestMetadataFor([factory.brandingSettings.logoFile], DEFAULT_IMAGE_THUMBNAIL)
+              .then(function (metadata) {
+                factory.brandingSettings.logoFileMetadata = metadata;
+              })
+              .catch(function (error) {
+                console.error('Could not load metadata for: ' + factory.brandingSettings.logoFile, error);
+              });            
+          } else {
+            factory.brandingSettings.logoFileMetadata = [];
```
Comment (alex-deaconu) [inline web/scripts/template-editor/components/services/svc-branding-factory.js]: @ezequielc is there any reason the metadata is part of the `brandingSettings` object? Can it be stored directly in `factory.logFileMetadata`? Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc looks good, a few issues I've found in my testing, not sure if they're on your list or not:
- After selecting a Logo from Storage, the title at the top becomes `Image -` and does not get updated until I leave the branding settings. 
- Sometimes I have to click the "Add Selected" link twice for the changes to apply (I think this is an existing issue)
- I can select multiple images for the logo, but only one gets applied. I think that's confusing.

Along with that there are some comments inline re: naming and refactoring.

I'm good with the PR as is since existing Image Component functionality is not affected, so you can address things or leave for subsequent PRs.

Thanks!

File: web/scripts/template-editor/components/services/svc-regular-image-factory.js (hunk 1 above)
Comment (ezequielc) [reply web/scripts/template-editor/components/services/svc-regular-image-factory.js]: @alex-deaconu Good suggestion, I will change it. Thanks

File: web/scripts/template-editor/services/svc-blueprint-factory.js (hunk 2 above)
Comment (ezequielc) [reply web/scripts/template-editor/services/svc-blueprint-factory.js]: @alex-deaconu Makes sense, I will take another look at this and at least rename it. Thanks

File: web/scripts/template-editor/components/services/svc-branding-factory.js (hunk 3 above)
Comment (ezequielc) [reply web/scripts/template-editor/components/services/svc-branding-factory.js]: @alex-deaconu This piece of code definitely needs improvement. I added as is so we could validate the functionality but we need to refactor and properly handle failures. I am planning to work on that in a separate PR. Thanks

File: web/scripts/template-editor/components/services/svc-branding-factory.js (hunk 4 above)
Comment (ezequielc) [reply web/scripts/template-editor/components/services/svc-branding-factory.js]: @alex-deaconu Very good suggestion. But same as above, I will handle that in another PR. Thanks!

File: web/scripts/template-editor/components/services/svc-branding-factory.js (hunk 3 above)
Comment (alex-deaconu) [reply web/scripts/template-editor/components/services/svc-branding-factory.js]: @ezequielc sounds good, thanks!

File: web/scripts/template-editor/components/services/svc-regular-image-factory.js (hunk 1 above)
Comment (alex-deaconu) [reply web/scripts/template-editor/components/services/svc-regular-image-factory.js]: @ezequielc thanks for updating 👍

**Comment by ezequielc** [comment]
@alex-deaconu I updated the image factory name as suggested. The other changes we can work in the next PRs. 
Re the issues, I have them listed in our Remaining Tasks card, except the "Add Selected" issue, which I am adding so we can pursue soon. There are some other minor problems that I identified and we need to correct before releasing this to users. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc sounds good, I'm good with the above. Thanks!

**Pull request merged**